### PR TITLE
PCHR-3342: Style Users List Page

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
As part of this PR(and https://github.com/compucorp/civihr-employee-portal/pull/444) the User List page in SSP is styled.

### List page
**Desktop**
desktop

**Mobile**
mobile

### Action Page
**Desktop**
2018-02-22 at 7 27 pm

**Mobile**
2018-02-22 at 7 27 pm

## Technical Details
1. `"Drupal", "location"` has been added to the `package.json` as they are global variables.
2. In `template.php`, UI Datepicker library files has been added using `drupal_add_library('system', 'ui.datepicker');`.
4. Styles has been added to `drupal/_view.scss`, `_table.scss` and `drupal/_form.scss` files.
3. And the styles which are not part of the above category , has been put into newly created`_users-list.scss`.

## Comments
The mobile datepicker is not implemented due to time crunch.

### CODE REVIEW

File: civihr_default_theme/assets/js/civihr_default_theme.script.js [hunk 1]
```diff
@@ -113,7 +114,7 @@
     });
   };
 
-  Drupal.civihr_theme.initMobileNav = function() {
+  Drupal.civihr_theme.initMobileNav = function () {
```
Comment (reneolivo) [inline civihr_default_theme/assets/js/civihr_default_theme.script.js]: missing documentation

File: civihr_default_theme/assets/js/civihr_default_theme.script.js [hunk 2]
```diff
@@ -310,4 +311,21 @@
     $('#' + this.id + '-' + 'day').val(dateValues.date).trigger('change');
     $('#' + this.id + '-' + 'year').val(dateValues.year).trigger('change');
   }
+
+  /**
+   * Do the stuff related to Users List Page
```
Comment (reneolivo) [inline civihr_default_theme/assets/js/civihr_default_theme.script.js]: `Initializes the date pickers in the Users List page`

File: civihr_default_theme/assets/js/civihr_default_theme.script.js [hunk 3]
```diff
@@ -310,4 +311,21 @@
...
+   */
+  function usersListPage () {
+    var isUsersListPage = $('.page-users-list').length;
```
Comment (reneolivo) [inline civihr_default_theme/assets/js/civihr_default_theme.script.js]: Can we do one of two things here:

1. rename this function to `onUsersListPage`, or
2. refactor whatever calls this function to this:
```js
$('.page-users-list').length && usersListPage();
```
And remove this condition from here.

I'm suggesting this so it's more clear that these changes only occur for the users list page and we are not implying that they would run for every page.

File: civihr_default_theme/scss/drupal/_view.scss [hunk 4]
```diff
@@ -36,30 +36,57 @@
...
+.view-full-hr-documents,
+.view-users-list {
   background: none;
   padding: 0;
 
   > .view-content {
     @include clearfix;
 
-    background: #ffffff;
+    background: #fff;
```
Comment (reneolivo) [inline civihr_default_theme/scss/drupal/_view.scss]: `$crm-white`

File: civihr_default_theme/scss/drupal/_view.scss [hunk 5]
```diff
@@ -36,30 +36,57 @@
...
   }
 
   > .view-filters {
-    background: #ffffff;
+    background: #fff;
```
Comment (reneolivo) [inline civihr_default_theme/scss/drupal/_view.scss]: `$crm-white`

**Review by reneolivo — CHANGES_REQUESTED** [review]

File: civihr_default_theme/assets/js/civihr_default_theme.script.js (hunk 3 above)
Comment (deb1990) [reply civihr_default_theme/assets/js/civihr_default_theme.script.js]: As discussed, it is not implemented.

File: civihr_default_theme/template.php [hunk 6]
```diff
@@ -406,6 +406,7 @@ function civihr_default_theme_preprocess_page(&$variables) {
   // Adds the theme path to Javascript variable.
   $path = drupal_get_path('theme', 'civihr_default_theme');
   drupal_add_js(array('civihr_default_theme' => array('path' => $path)), 'setting');
+  drupal_add_library('system', 'ui.datepicker');
```
Comment (tunbola) [inline civihr_default_theme/template.php:409]: Did you eventually find out why this was not included for onboarding but needs to be explicitly included for this view?
Is there something that was done differently there?

Comment (deb1990) [reply civihr_default_theme/template.php:409]: Yes, in onboarding we are using webforms which includes datepicker in `sites/all/modules/contrib/webform/components/date.inc`

**Review by reneolivo — APPROVED** [review]

**Review by tunbola — APPROVED** [review]

**Pull request merged**